contracts/src/LightClient.sol

```diff
@@ -341,4 +381,78 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice check if more than threshold blocks passed since the last state update before
+    /// L1 Block Number
+    /// @param blockNumber The L1 block number
+    /// @param threshold The number of blocks updates to this contract is allowed to lag behind
+    function lagOverEscapeHatchThreshold(uint256 blockNumber, uint256 threshold)
+        public
+        view
+        returns (bool)
+    {
+        uint256 updatesCount = stateUpdateBlockNumbers.length;
+
+        // Handling Edge Cases
+        // Edgecase 1: The block is in the future or in the past before HotShot was live
+        if (blockNumber > block.number || updatesCount < 3) {
```

Comment:
If the block is in the future, doen't that mean hotshot has already down?

Reply:
hmm, if the current ethereum block is 10 and the user requested for block 12, i'd say that we can't say whether our light client contract is missing an update / hotshot is down because we don't have enough data points to be able to assert that.

Reply:
IMO it's not meaningful to ask if the light client contract lagged behind at an Ethereum block number in the future so I think it's correct to throw an error here.